addons/io_scene_gltf2/blender/exp/gltf2_blender_generate_extras.py

```diff
@@ -16,6 +16,37 @@
 import bpy
 from io_scene_gltf2.blender.com import gltf2_blender_json
 
+def to_json_compatible(value):
+    """Make a value (usually a custom property) compatible with json"""
+
+    if isinstance(value, bpy.types.ID):
+        return value
+
+    elif isinstance(value, str):
+        return value
+
+    elif isinstance(value, (int, float)):
+        return value
+
+    # for list classes
+    elif isinstance(value, list):
+        value = list(value)
+        # make sure contents are json-compatible too
+        for index in range(len(value)):
+            value[index] = to_json_compatible(value[index])
+        return value
+
+    # for IDPropertyArray classes
+    elif hasattr(value, "to_list"):
+        value = value.to_list()
+        return value
+
+    elif hasattr(value, "to_dict"):
+        value = value.to_dict()
+        if gltf2_blender_json.is_json_convertible(value):
+            return value;
```

Comment:
for [PEP8](https://www.python.org/dev/peps/pep-0008/) conformity, please remove the `;`